modules/lvm2/udiskslinuxvdovolume.c

```diff
@@ -281,11 +304,135 @@ handle_enable_deduplication (UDisksVDOVolume       *_volume,
                                            FALSE /* compression */, TRUE /* deduplication */, options);
 }
 
+static void _vdo_resize (UDisksLinuxLogicalVolumeObject *object,
+                         GDBusMethodInvocation          *invocation,
+                         guint64                         new_size,
+                         GVariant                       *options)
+{
+  GError *error = NULL;
+  UDisksDaemon *daemon = NULL;
+  UDisksLinuxVolumeGroupObject *group_object = NULL;
+  uid_t caller_uid;
+  LVJobData data;
+
+  if (!common_setup (object, invocation, options,
+                     N_("Authentication is required to resize a VDO volume"),
+                     &daemon, &caller_uid))
+    goto out;
+
+  group_object = udisks_linux_logical_volume_object_get_volume_group (object);
+  data.vg_name = udisks_linux_volume_group_object_get_name (group_object);
+  data.lv_name = udisks_linux_logical_volume_object_get_name (object);
+  data.new_lv_size = new_size;
+
+  data.resize_fs = FALSE;
+  data.force = FALSE;
+  g_variant_lookup (options, "resize_fsys", "b", &(data.resize_fs));
+  g_variant_lookup (options, "force", "b", &(data.force));
+
+  if (!udisks_daemon_launch_threaded_job_sync (daemon,
+                                               UDISKS_OBJECT (object),
+                                               "lvm-lvol-resize",
+                                               caller_uid,
+                                               lvresize_job_func,
+                                               &data,
+                                               NULL, /* user_data_free_func */
+                                               NULL, /* GCancellable */
+                                               &error))
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             UDISKS_ERROR,
+                                             UDISKS_ERROR_FAILED,
+                                             "Error resizing logical volume: %s",
+                                             error->message);
+      g_clear_error (&error);
+      goto out;
+    }
+
+ out:
+  g_clear_object (&object);
+}
+
+static gboolean
+handle_resize_logical (UDisksVDOVolume       *_volume,
+                       GDBusMethodInvocation *invocation,
+                       guint64                new_size,
+                       GVariant              *options)
+{
+  UDisksLinuxLogicalVolumeObject *object = NULL;
+  GError *error = NULL;
+
+  object = udisks_daemon_util_dup_object (_volume, &error);
+    if (object == NULL)
+      {
+        g_dbus_method_invocation_take_error (invocation, error);
+        goto out;
+      }
+
+  _vdo_resize (object, invocation, new_size, options);
+
+  udisks_vdo_volume_complete_resize_logical (_volume, invocation);
```

Comment:
This will overwrite the error set through `g_dbus_method_invocation_return_error()`. Or rather will fail as the D-Bus message may have already been sent (a duplicate return message would get sent and the client side would ignore it hopefully).

You probably need to return success indicator from `_vdo_resize()` and act accordingly.

Reply:
Fixed.